src/devices/Is31fl3730/Is31fl3730.cs

```diff
@@ -0,0 +1,447 @@
+
+        /// <summary>
+        /// Disable all LEDs.
+        /// </summary>
+        public void DisableAllLeds()
+        {
+            if (_matrix1Enabled)
+            {
+                Write(MATRIX_1_REGISTER, _disable_all_leds_data);
+            }
+
+            if (_matrix2Enabled)
+            {
+                Write(MATRIX_2_REGISTER, _disable_all_leds_data);
+            }
+
+            WriteUpdateRegister();
+        }
+
+        /// <summary>
+        /// Set display mode. Call before Initialize method.
+        /// </summary>
+        public void SetDisplayMode(bool matrixOne, bool matrixTwo)
+        {
+            _matrix1Enabled = matrixOne;
+            _matrix2Enabled = matrixTwo;
+
+            _configurationValue |= (matrixOne, matrixTwo) switch
+            {
+                (true, true) => DISPLAY_MATRIX_BOTH,
+                (true, _) => DISPLAY_MATRIX_ONE,
+                (_, true) => DISPLAY_MATRIX_TWO,
+                _ => throw new Exception("Invalid input.")
+            };
+        }
+
+        /// <summary>
+        /// Set matrix mode. Call before Initialize method.
+        /// </summary>
+        public void SetMatrixMode(MatrixMode mode)
+        {
+            _configurationValue |= mode switch
+            {
+                MatrixMode.Size5x11 => MATRIX_5x11,
+                MatrixMode.Size6x10 => MATRIX_6x10,
+                MatrixMode.Size7x9 => MATRIX_7x9,
+                MatrixMode.Size8x8 => MATRIX_8x8,
+                _ => throw new Exception("Invalid input.")
+            };
+        }
+
+        /// <summary>
+        /// Reset device.
+        /// </summary>
+        public void Reset()
+        {
+            // Reset device
+            Shutdown(true);
+            Thread.Sleep(10);
+            Shutdown(false);
+        }
+
+        /// <summary>
+        /// Set the shutdown mode.
+        /// </summary>
+        /// <param name="shutdown">Set the showdown mode. `true` sets device into shutdown mode. `false` sets device into normal operation.</param>
+        public void Shutdown(bool shutdown)
+        {
+            // mode values
+            // 0 = normal operation
+            // 1 = shutdown mode
+            if (shutdown)
+            {
+                _configurationValue |= SHUTDOWN;
+            }
+            else
+            {
+                _configurationValue &= ~SHUTDOWN;
+            }
+
+            WriteUpdateRegister();
+        }
+
+        /// <inheritdoc/>
+        public void Dispose()
+        {
+            _i2cDevice?.Dispose();
+            _i2cDevice = null!;
+        }
+
+        private void WriteConfiguration()
+        {
+            Write(CONFIGURATION_REGISTER, (byte)_configurationValue);
+        }
+
+        private void Write(byte address, byte[] value)
+        {
+            byte[] data = new byte[value.Length + 1];
+            data[0] = address;
+            value.CopyTo(data.AsSpan(1));
+            _i2cDevice.Write(data);
+        }
```

Comment:
```suggestion
        private void Write(byte address, ReadOnlySpan<byte> value)
        {
            Debug.Assert(value.Length < 256);
            
            Span<byte> data = stackalloc byte[value.Length + ];
            data[0] = address;
            value.CopyTo(data.Slice(1));
            _i2cDevice.Write(data);
        }
```
Avoids the temp. allocation.